tests/test_gpu_common.py

```diff
@@ -1128,6 +1128,44 @@ def test_xcor_from_saved(self, opt, opt_options, gpu_fit):
 
         assert np.all(g.data == 30)
 
+    def test_devicecreate_forward(self):
```

Comment:
on second thought, I think we need to check codegen. Something very minimal -- eg to make sure that there's indeed a create(...) and not a mapto(). 

```
if openacc:
    assert 'create(u)' in str(...)
elif openmp
    assert '...'
```

(it's target-language specific; you can pull the target language from `configuration`)